docs/api.md

```diff
@@ -0,0 +1,550 @@
+            "name": "webhook",
+            "index": 0,
+            "type": "WEBHOOK",
+            "enabled": true,
+            "webhook_host": "http://service:42135/hook",
+            "webhook_method": "POST",
+            "webhook_data": "{\n    \"torrent_id\": {{.TorrentID}},\n    \"apikey\": \"redacted.apikey\",\n    \"maxsize\": 2147483648,\n    \"record_labels\": \"Label1,Label2\"\n}",
+            "webhook_expect_status": 200
+        }
+    ]
+}'
+```
+
+## Indexers
+
+### Fetch all indexers
+
+Retrieve a list of all indexers configured in your autobrr instance.
+
+```bash
+curl -X GET 'http://127.0.0.1:7474/api/indexer' -H 'X-API-Token: AUTOBRR_API_KEY' | jq '.[] | {id, name, enabled}'
+```
+
+### Enable or disable an indexer
+
+Toggle the status of a specific indexer.
+
+:::caution Note
+
+There is a known issue with this endpoint. It may require sending a request twice to ensure a successful change.
+This behavior is observed in the web UI as well.
+
+:::
+
+```bash
+curl -X PATCH 'http://127.0.0.1:7474/api/indexer/31/enabled' -H 'X-API-Token: AUTOBRR_API_KEY' \
+     -d '{"enabled": true}'
+```
+
+## IRC Networks
+
+### List all networks
+
+```bash
+curl -X GET 'http://127.0.0.1:7474/api/irc' -H 'X-API-Token: AUTOBRR_API_KEY' | jq '.[] | {id, name, healthy}'
+```
+
+### Restart a network
+
+```bash
+curl -X GET 'http://127.0.0.1:7474/api/irc/network/5/restart' -H 'X-API-Token: AUTOBRR_API_KEY'
+```
+
+## Feeds
+
+### Fetch all feeds
+
+Retrieve a list of all feeds available on your autobrr instance.
+
+```bash
+curl -X GET 'http://127.0.0.1:7474/api/feeds' -H 'X-API-Token: AUTOBRR_API_KEY' | jq '.[] | {id, name, enabled}'
+```
+
+### Enable or disable a feed
+
+Toggle the status of a specific feed.
+
+```bash
+curl -X PATCH 'http://127.0.0.1:7474/api/feeds/8/enabled' -H 'X-API-Token: AUTOBRR_API_KEY' \
+     -d '{"enabled": true}'
+```
+
+### Clear the cache of a feed
+
+```bash
+curl -X DELETE 'http://127.0.0.1:7474/api/feeds/8/cache' -H 'X-API-Token: AUTOBRR_API_KEY'
+```
+
+## Download clients
+
+### List all download clients
+
+```bash
+curl -X GET 'http://127.0.0.1:7474/api/download_clients' -H 'X-API-Token: AUTOBRR_API_KEY' | jq
+```
+
+### Add a new download client
+
+```bash title="qBittorrent"
+curl -X POST 'http://127.0.0.1:7474/api/download_clients' -H 'X-API-Token: AUTOBRR_API_KEY' \
+-d '{
+  "name": "qbit",
+  "type": "QBITTORRENT",
+  "enabled": true,
+  "host": "http://172.17.0.1:10963",
+  "port": 0,
+  "tls": false,
+  "tls_skip_verify": false,
+  "username": "username",
+  "password": "password",
+  "settings": {
+    "basic": {
+      "auth": true,
+      "username": "username",
+      "password": "password"
+    },
+    "rules": {
+      "enabled": true,
+      "max_active_downloads": 1,
+      "ignore_slow_torrents": true,
+      "ignore_slow_torrents_condition": "MAX_DOWNLOADS_REACHED",
+      "download_speed_threshold": 10000,
+      "upload_speed_threshold": 400
+    }
+  }
+}'
+```
+
```

Comment:
You provided a Deluge example for updating a existing download client..
Do you happen to have an example for creation here too?